### PR TITLE
feat(pubsub): show ack throughput in benchmark

### PR DESCRIPTION
**Pull request by coryan**
Report (separately) the throughput for publishing messages and processing
acks. This can show when the acks are not being processed fast enough,
or when the messages are queued.

I refactored the code because it was hard to follow what was going on
with all the lambdas. I reduced contention in the benchmark by tracking
pending messages for each sending thread separately.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5439)

### CODE REVIEW

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/benchmarks/throughput.cc [hunk 1]
```diff
@@ -203,6 +204,99 @@ void PrintResult(Config const& config, std::string const& operation,
             << std::endl;
 }
 
+std::atomic<std::int64_t> send_count{0};
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/pubsub/benchmarks/throughput.cc:208]: Done.

File: google/cloud/pubsub/benchmarks/throughput.cc [hunk 2]
```diff
@@ -203,6 +204,99 @@ void PrintResult(Config const& config, std::string const& operation,
...
+    auto gen = google::cloud::internal::DefaultPRNG(std::random_device{}());
+    auto const data = google::cloud::internal::Sample(
+        gen, static_cast<int>(config_.payload_size), "0123456789");
+
+    using std::chrono::duration_cast;
+    using std::chrono::steady_clock;
+
+    auto const start = std::chrono::steady_clock::now();
+    for (std::int64_t i = 0; true; ++i) {
+      if (WaitUntilCanPublish()) break;  // returns true on shutdown
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/pubsub/benchmarks/throughput.cc]: Done.
